volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -204,138 +223,41 @@ def get_module_info_for_module(
-
-    @classmethod
-    def _get_sysfs_modules(
-        cls, context: interfaces.context.ContextInterface, kernel_module_name: str
-    ) -> List[ModuleInfo]:
-        """
-        Gather the `module` instances from sysfs
-        """
-        kernel = context.modules[kernel_module_name]
-
-        sysfs_modules: dict = cls.get_kset_modules(context, kernel_module_name)
-
-        for m_offset in sysfs_modules.values():
-            yield kernel.object(object_type="module", offset=m_offset, absolute=True)
-
-    @classmethod
-    def _validated_sources(cls, caller_wanted_sources) -> List[str]:
-        """
-        Called by `run_modules_scanners` to validate the caller supplied sources list
-        An exception is thrown if an empty source list is given or a list containing an invalid source
-        """
-        if not caller_wanted_sources:
-            raise ValueError("`caller_wanted_sources` must have at least one source.")
-
-        if (
-            len(caller_wanted_sources) == 1
-            and caller_wanted_sources[0] == Modules.source_hidden_identifier
-        ):
+        if not caller_wanted_gatherers:
             raise ValueError(
-                f"{Modules.source_hidden_identifier} cannot be the only source or there is nothing to compare against."
+                "`caller_wanted_gatherers` must have at least one gatherer."
             )
 
-        wanted_sources = []
-
-        for source in caller_wanted_sources:
-            if source not in Modules.all_sources_identifier:
+        for gatherer in caller_wanted_gatherers:
+            if gatherer not in ModuleGatherers.all_gatherers_identifier:
                 raise ValueError(
-                    f"Invalid source sent through `caller_wanted_sources`: {source}"
+                    f"Invalid gatherer sent through `caller_wanted_gatherers`: {gatherer}"
                 )
 
-            wanted_sources.append(source)
-
-        return wanted_sources
-
     @classmethod
     def run_modules_scanners(
         cls,
         context: interfaces.context.ContextInterface,
         kernel_module_name: str,
-        caller_wanted_sources: List[str],
+        caller_wanted_gatherers: List[ModuleGathererInterface],
         flatten: bool = True,
-    ) -> Dict[str, List[ModuleInfo]]:
+    ) -> Dict[ModuleGathererInterface, List[ModuleInfo]]:
         """Run module scanning plugins and aggregate the results. It is designed
         to not operate any inter-plugin results triage.
 
         Rules for `caller_wanted_sources`:
 
-            If `Modules.all_sources_identifier` is specified then every source will be populated
+            If `ModuleGathers.all_gathers_identifier` is specified then every source will be populated
 
-            If `Modules.source_hidden_identifier` is in the list, then at least one other sources must be
+            If `ModuleGathers.Scanner` is in the list, then at least one other sources must be
```

Comment:
Getting bad vibes from this being conditional, I'll have to see where it's going...

Reply:
I don't think this comment is accurate?  I can't see anything in the Scanner that requires anything else?  It looks like it always scans all modules, just as expected?

Reply:
Nice catch, that was a stale comment. about to push a commit with it gone.